### PR TITLE
[pipeline API] Let pipeline src use current caps when it's given

### PR DESCRIPTION
**Pull request by anyj0527**
- If current caps is given, use it rather than allowed caps.
- Use caps "image/png" when the pipeline is "appsrc caps=image/png ! pngdec ! ... "
- Add a test for decoding pngfile using appsrc (C API and android JAVA)
- It check the equality of decoded orange.png using simple pipline
  with raw file (orange.raw)

This resolves https://github.com/nnstreamer/nnstreamer/issues/3753

Self evaluation:
Build test: [X]Passed [ ]Failed [ ]Skipped
Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

### CODE REVIEW

File: tests/capi/unittest_capi_inference.cc [hunk 1]
```diff
@@ -1437,6 +1438,175 @@ TEST (nnstreamer_capi_src, callback_invalid_param_02_n)
...
+  fd = open (orange_png_file, O_RDONLY);
+  EXPECT_TRUE (fd >= 0);
+
+  pipeline = g_strdup_printf (
+    "appsrc name=srcx caps=image/png ! pngdec ! videoconvert ! videoscale ! video/x-raw,format=RGB,width=224,height=224,framerate=0/1 ! tensor_converter ! tensor_sink name=sinkx sync=false async=false"
+  );
+
+  /* construct pipeline */
+  status = ml_pipeline_construct (pipeline, NULL, NULL, &handle);
+  EXPECT_EQ (status ,ML_ERROR_NONE);
```
Comment (gichan-jang) [inline tests/capi/unittest_capi_inference.cc:1532]: Minor style change:
```suggestion
  EXPECT_EQ (status, ML_ERROR_NONE);
```
+ L1536

Comment (anyj0527) [reply tests/capi/unittest_capi_inference.cc]: Thanks! I fixed the code.

File: tests/capi/unittest_capi_inference.cc [hunk 2]
```diff
@@ -1437,6 +1438,175 @@ TEST (nnstreamer_capi_src, callback_invalid_param_02_n)
...
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  orange_raw_file = g_build_filename (
+      root_path, "tests", "test_models", "data", "orange.raw", NULL);
+  ASSERT_TRUE (g_file_test (orange_raw_file, G_FILE_TEST_EXISTS));
+
+  fd = open (orange_raw_file, G_FILE_TEST_EXISTS);
```
Comment (jaeyun-jung) [inline tests/capi/unittest_capi_inference.cc:1462]: plz consider to get file content with g_file_get_content()

Comment (anyj0527) [reply tests/capi/unittest_capi_inference.cc]: Cool, the `g_file_get_contents` is way more safer. I updated the code with it, Thanks

**Review by jaeyun-jung — APPROVED** [review]

**Review by gichan-jang — APPROVED** [review]
LGTM!

**Pull request merged**